pages/index.js

```diff
@@ -127,6 +164,12 @@ export default function HomePage({ onToggleFavorite, favorites }) {
                   <StyledListItem
                     key={index}
                     onMouseDown={() => handleLastSearchClick(event, search)}
+                    onMouseEnter={() => setCurrentIndex(index)}
+                    onMouseLeave={() => setCurrentIndex(-1)}
+                    tabIndex={0}
+                    style={{
+                      fontWeight: currentIndex === index ? 600 : "",
+                    }}
```

Comment:
You might want to write this as a prop for the styled component (something like `$highlight`), and the component itself would decide how to highlight the item – just to keep things consistent.